scr/main/java/bittercode/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.bittercode.service.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import javax.servlet.http.HttpSession;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.constant.db.UsersDBConstants;
+import com.bittercode.model.StoreException;
+import com.bittercode.model.User;
+import com.bittercode.model.UserRole;
+import com.bittercode.service.UserService;
+import com.bittercode.util.DBUtil;
+
+public class UserServiceImpl implements UserService {
+
+    private static final String registerUserQuery = "INSERT INTO " + UsersDBConstants.TABLE_USERS
+            + "  VALUES(?,?,?,?,?,?,?,?)";
+
+    private static final String loginUserQuery = "SELECT * FROM " + UsersDBConstants.TABLE_USERS + " WHERE "
+            + UsersDBConstants.COLUMN_USERNAME + "=? AND " + UsersDBConstants.COLUMN_PASSWORD + "=? AND "
+            + UsersDBConstants.COLUMN_USERTYPE + "=?";
+
+    @Override
+    public User login(UserRole role, String email, String password, HttpSession session) throws StoreException {
+        Connection con = DBUtil.getConnection();
+        PreparedStatement ps;
+        User user = null;
+        try {
+            String userType = UserRole.SELLER.equals(role) ? "1" : "2";
+            ps = con.prepareStatement(loginUserQuery);
+            ps.setString(1, email);
+            ps.setString(2, password);
+            ps.setString(3, userType);
+            ResultSet rs = ps.executeQuery();
+            if (rs.next()) {
+                user = new User();
+                user.setFirstName(rs.getString("firstName"));
+                user.setLastName(rs.getString("lastName"));
+                user.setPhone(rs.getLong("phone"));
+                user.setEmailId(email);
+                user.setPassword(password);
+                session.setAttribute(role.toString(), user.getEmailId());
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return user;
+    }
+
+    @Override
+    public boolean isLoggedIn(UserRole role, HttpSession session) {
+        if (role == null)
+            role = UserRole.CUSTOMER;
+        return session.getAttribute(role.toString()) != null;
+    }
+
+    @Override
+    public boolean logout(HttpSession session) {
+        session.removeAttribute(UserRole.CUSTOMER.toString());
+        session.removeAttribute(UserRole.SELLER.toString());
+        session.invalidate();
+        return true;
+    }
+
+    @Override
+    public String register(UserRole role, User user) throws StoreException {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2438181340>#16fdcd</a><span> - 10/25/2024, 04:08 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The register method obtains a new database connection for each registration, which may not scale well under high load.
</div>
<div id="fix">
<b>Fix:</b> Implement a connection pool and use try-with-resources for automatic resource management.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -69,28 +69,27 @@
  @Override
  public String register(UserRole role, User user) throws StoreException {
      String responseMessage = ResponseCode.FAILURE.name();
 -    Connection con = DBUtil.getConnection();
 -    try {
 -        PreparedStatement ps = con.prepareStatement(registerUserQuery);
 +    try (Connection con = DBUtil.getConnection();
 +         PreparedStatement ps = con.prepareStatement(registerUserQuery)) {
          ps.setString(1, user.getEmailId());
          ps.setString(2, user.getPassword());
          ps.setString(3, user.getFirstName());
          ps.setString(4, user.getLastName());
          ps.setString(5, user.getAddress());
          ps.setLong(6, user.getPhone());
          ps.setString(7, user.getEmailId());
          int userType = UserRole.SELLER.equals(role) ? 1 : 2;
          ps.setInt(8, userType);
          int k = ps.executeUpdate();
          if (k == 1) {
              responseMessage = ResponseCode.SUCCESS.name();
 -            ;
          }
 -    } catch (Exception e) {
 -        responseMessage += " : " + e.getMessage();
 -        if (responseMessage.contains("Duplicate"))
 -            responseMessage = "User already registered with this email !!";
 -        e.printStackTrace();
 +    } catch (SQLException e) {
 +        if (e.getMessage().contains("Duplicate")) {
 +            responseMessage = "User already registered with this email !!";
 +        } else {
 +            throw new StoreException(ResponseCode.DATABASE_ERROR, e);
 +        }
      }
      return responseMessage;
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged